### DISCUSSION TITLE
Unauthorised error not redirecting for the first time in oathkeeper

### DISCUSSION THREAD

**Discussion by vsumit89** [ory/oathkeeper#985] [Q&A] [answered]
I am encountering a unauthorised error in oathkeeper.
In my application when I logout and again try to hit the protected URL, for the first time it throws Access Credentials are Invalid error and then does not redirect to the login page even when I have a handler for unauthorised error. But when I reload the page again it throws the same error but this time it redirects to the login page.
These are my oathkeeper rules.
```
handlers:
    redirect:
      enabled: true
      config:
        to: http://127.0.0.1:4455/.factly/kavach/web/auth/login
        when:
          -
            error:
              - unauthorized
              - forbidden
            request:
              header:
                accept:
                  - text/html
    json:
      enabled: true
      config:
        verbose: true
```

**Comment by vinckr** [comment, marked as answer]
Check out our [Oathkeeper examples](https://github.com/ory/examples/tree/master/oathkeeper), maybe those can be of help. If they don't help let me know.

**Reply by vsumit89** [reply]
Hey @vinckr, I went through the examples and I have exactly the same implementation but it doesn't work the way it should.
This is for reference from oathkeeper examples - 
```
handlers:
    redirect:
      enabled: true
      config:
        to: http://127.0.0.1:4455/login
        when:
          - error:
              - unauthorized
              - forbidden
            request:
              header:
                accept:
                  - text/html
```
When I logout from the application and hit a protected URL without having the  `cookie_session`, instead of throwing unauthorized error and redirecting to the login page, oathkeeper lets it pass through for the first time, but when I reload it, it redirects.
This is the protected URL - `http://127.0.0.1:4455/.factly/dega/studio/<.*>`
This is the URL to which I am redirecting - `http://127.0.0.1:4455/.factly/kavach/web/auth/login`

**Reply by Intevel** [reply]
Hey @vsumit89! Did you fixed this?

**Reply by vsumit89** [reply]
@Intevel No I wasn't able to.

**Reply by fswu0831** [reply]
Hi @Intevel  @vsumit89 
I'm confronting same issue. Dis you fixed this?
I tried to use Oathkeeper v0.40.8 and v0.39.4

**Reply by Intevel** [reply]
Nope
